### ISSUE TITLE
Null of comlex type in List<T> and Dictionary<TKey, TValue> deserialized as non-null object

### ISSUE THREAD

**Issue by tchekjunior** [msgpack/msgpack-cli#211]
hi,
i have some problems to unpack list with null elements.
how to fix it please
```
public class SimpleItem
{
    public int Id;
    public SimpleItem()
    {
}
}
var items= new List<SimpleItem>();
                    items.Add(null);
items[0] = new SimpleItem();

var stream = new MemoryStream();
var context = new SerializationContext();
var serializer = context.GetSerializer<SimpleItem>();

serializer .Pack(stream, items);
stream.Position = 0;
var cloneItems = serializer .Unpack(stream);
```
cloneItems null item in list is not null, why?

**Comment by yfakariya** [comment]
Sorry, it is a critical bug. Thank you for reporting!

**Comment by tchekjunior** [comment]
hi, thank, i can investigate if you help me, i want to fix it quilkly.
i use build dotNet3.5.
I have meet order dysfonctionnement ,I write other issue after.
but good job is very fast serializer.

**Comment by tchekjunior** [comment]
Thank you very much for your reactivity to this problem.  i try it now. good job

**Comment by yfakariya** [comment]
Thank you. I will release it as 0.8.1 soon.

**Comment by yfakariya** [comment]
BTW, your "order dysfonctionnement" is related to #71?

**Comment by yfakariya** [comment]
Fixed in 7e73aff
